File: numeric/primes500/primes.py

def get_primes():
    primes = [2]
    n = 3

    for j in range(500):
        primes.append(n)

        is_prime = False
        while not is_prime:
            n += 2

            q = 9999
            r = 1
            t = 0
            k = 1
            while (r != 0) and (q > t):
                t = primes[k]
                q = n / t
                r = n % t
                k += 1
            is_prime = (r != 0) and (q <= t)
    return primes

def print_primes(primes):
    print("First Five Hundred Primes")
    for i in range(50):
        print(f"     {primes[i]:04d} {primes[i + 50]:04d}" +
              f" {primes[i + 100]:04d} {primes[i + 150]:04d}" +
              f" {primes[i + 200]:04d} {primes[i + 250]:04d}" +
              f" {primes[i + 300]:04d} {primes[i + 350]:04d}" +
              f" {primes[i + 400]:04d} {primes[i + 450]:04d}")

print_primes(get_primes())

File: numeric/primes500/primes.ts

function get_primes(): Array<number> {
    let primes = [2];
    var n = 3;

    for (var j = 1; j < 500; ++j) {
        primes.push(n);

        var is_prime = false;
        while (!is_prime) {
            n += 2;

            var q = 9999;
            var r = 1;
            var t = 0;
            for (var k = 1; (r != 0) && (q > t); ++k) {
                t = primes[k];
                q = n / t;
                r = n % t;
            }

            is_prime = (r != 0) && (q <= t);
        }
    }

    return primes;
}

function print_primes(primes: Array<number>) {
    console.log("First Five Hundred Primes");
    var pad = (n: number) => n.toString().padStart(4, '0');
    for (var n = 0; n < 50; ++n) {
        console.log(`     ${pad(primes[n])} ${pad(primes[n + 50])}`
            + ` ${pad(primes[n + 100])} ${pad(primes[n + 150])}`
            + ` ${pad(primes[n + 200])} ${pad(primes[n + 250])}`
            + ` ${pad(primes[n + 300])} ${pad(primes[n + 350])}`
            + ` ${pad(primes[n + 400])} ${pad(primes[n + 450])}`);
    }
}

print_primes(get_primes());
